### PR TITLE
fix: add eslint fixes, eslint rules, dev dependencies.

### PR DESCRIPTION
**Pull request by tangollama**

### CODE REVIEW

File: nerdlets/e2m-gui-nerdlet/account-details.js [hunk 1]
```diff
@@ -1,110 +1,109 @@
...
+          </div>
+        )}
+        <div className="refreshcache">
+          You are seeing cached data. To refresh click{' '}
+          <span onClick={() => this.props.refreshCardinality()}>here</span>.
+        </div>
+      </div>
+    );
+  }
 }
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/account-details.js:100]: Diff lost sync, check files.

File: nerdlets/e2m-gui-nerdlet/detail-components/attributes.js [hunk 2]
```diff
@@ -1,23 +1,24 @@
-import React from "react";
-import PropTypes from "prop-types";
+import React from 'react';
+import PropTypes from 'prop-types';
 
 import { parseNRQL } from '../../util/misc';
 
-const MetricAttributes = ({ nrql, accountId }) => {
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/detail-components/attributes.js:6]: Motivation?

File: nerdlets/e2m-gui-nerdlet/detail-components/attributes.js [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
+        ))}
+      </ul>
+    </div>
+  );
 };
 
 MetricAttributes.propTypes = {
-	accountId: PropTypes.number.isRequired,
-	facets: PropTypes.array
+  nrql: PropTypes.string
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/detail-components/attributes.js:21]: Semantics?

File: nerdlets/e2m-gui-nerdlet/detail-components/rule-history.js [hunk 4]
```diff
@@ -1,60 +1,60 @@
-import React from "react";
-import PropTypes from "prop-types";
+import React from 'react';
+import PropTypes from 'prop-types';
 
-import { NrqlQuery, NerdGraphQuery, Spinner } from "nr1";
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/detail-components/rule-history.js:4]: Unused?

File: nerdlets/e2m-gui-nerdlet/form-components/account-selector.js [hunk 5]
```diff
@@ -1,45 +1,48 @@
-import React from "react";
-import PropTypes from "prop-types";
-
-import SelectSearch from "react-select-search";
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Spinner } from 'nr1';
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/form-components/account-selector.js:3]: Previous Spinner?

File: nerdlets/e2m-gui-nerdlet/form-components/attribute-checkbox-selector.js [hunk 6]
```diff
@@ -1,67 +1,62 @@
-import React from "react";
-import PropTypes from "prop-types";
-import { Grid, GridItem, Checkbox, NerdGraphQuery, Spinner } from "nr1";
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/form-components/attribute-checkbox-selector.js:3]: Unused?

File: nerdlets/e2m-gui-nerdlet/form-components/attribute-dropdown-selector.js [hunk 7]
```diff
@@ -1,131 +1,124 @@
...
-			console.log(this.state.error);
-			return <div>Check console for error</div>;
-		} else if (this.state.fetching) {
-			return <Spinner />;
-		}
+  render() {
+    // Since multiple accounts are selected, the event types shown below are aggregated from
+    //      all accounts. Each individual account may not have the event types here.
+    if (this.state.error) {
+      console.log(this.state.error); // eslint-disable-line no-console
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/form-components/attribute-dropdown-selector.js:71]: Rationale?

File: nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js [hunk 8]
```diff
@@ -1,90 +1,88 @@
...
-			validationTimeout: null,
-			validating: false,
-			validated: null
-		});
-	}
+  clear() {
+    this.setState({
+      ruleDescription: null
+    });
+  }
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js:17]: Obscure?

File: nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js [hunk 9]
```diff
@@ -1,90 +1,88 @@
...
 
-	validRuleDescription(name) {
-		return true;
-	}
+  /* eslint-disable no-unused-vars */
+  validRuleDescription(name) {
+    return true;
+  }
+  /* eslint-enable */
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js:40]: Signature

File: nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js [hunk 10]
```diff
@@ -1,90 +1,88 @@
...
-									/[^a-zA-Z0-9\.\s ]/g,
-									""
-							  )
-					}
-				/>
-			</div>
-		);
-	}
+  render() {
+    /* TODO: Pull the disabledOrNotObj out to AddRule Component */
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js:53]: TODO

File: nerdlets/e2m-gui-nerdlet/form-components/section2.js [hunk 11]
```diff
@@ -1,146 +1,140 @@
...
+              type={Button.TYPE.PRIMARY}
+            >
+              Next Step
+            </Button>
+          )}
+        </div>
+      </div>
+    </div>
+  );
 };
```
Comment (WardCunningham) [inline nerdlets/e2m-gui-nerdlet/form-components/section2.js:118]: Have to stop checking here. Another meeting.

File: nerdlets/e2m-gui-nerdlet/detail-components/attributes.js (hunk 3 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/detail-components/attributes.js:21]: eslint check calls out unused PropTypes as well as props used without validation.

File: nerdlets/e2m-gui-nerdlet/detail-components/rule-history.js (hunk 4 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/detail-components/rule-history.js:4]: Correct.

File: nerdlets/e2m-gui-nerdlet/account-details.js (hunk 1 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/account-details.js:100]: Linting rules adjusted the format via `npm run eslint-fix`

File: nerdlets/e2m-gui-nerdlet/detail-components/attributes.js (hunk 2 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/detail-components/attributes.js:6]: `accountId` was unused.

File: nerdlets/e2m-gui-nerdlet/form-components/account-selector.js (hunk 5 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/form-components/account-selector.js:3]: The code may not have loaded b/c it was referenced in the code. If it's never executed, there's a chance that the inevitable error never occurred.

File: nerdlets/e2m-gui-nerdlet/form-components/attribute-checkbox-selector.js (hunk 6 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/form-components/attribute-checkbox-selector.js:3]: Correct

File: nerdlets/e2m-gui-nerdlet/form-components/attribute-dropdown-selector.js (hunk 7 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/form-components/attribute-dropdown-selector.js:71]: I made the assumption in the linting rules that any log of an error was an acceptable exception and added the appropriate rules.

File: nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js (hunk 8 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js:17]: Linting errors brought up that those two fields where unused.

File: nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js (hunk 9 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js:40]: I'm assuming that there's a future for that function and the related input. That said, the linting rules would disallow the name input that's unused.

File: nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js (hunk 10 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/form-components/rule-description-validator.js:53]: Maintained from the original code.

File: nerdlets/e2m-gui-nerdlet/form-components/section2.js (hunk 11 above)
Comment (tangollama) [reply nerdlets/e2m-gui-nerdlet/form-components/section2.js:118]: This one is pretty straightforward formatting cleanup.

**Pull request closed without merging**